### ISSUE TITLE
ListObjectsAsync giving me inconsistent results

### ISSUE THREAD

**Issue by mia01** [minio/minio-dotnet#418]
I'm writing some integration tests to test a service that uploads to a local dockerised minio instance. At one point in my test I am checking if things have been uploaded using ListObjectsAsync like this:

```
await minioClient.PutObjectAsync(bucketName, $"folder/myObject", fileStream, fileStream.Length, ".pdf");

List<string> objects= new List<string>();

IObservable<Item> observable = minioClient.ListObjectsAsync(bucketName, "folder/");
                var subscription = observable.Subscribe(
                    item => objects.Add(item.Key),
                    ex => throw ex
                );

observable.Wait();
subscription.Dispose();
return objects;
```

Occasionally I get an empty objects list and at other times I will get the expected results with "folder/myObject.pdf" listed. Am I doing anything wrong here? Everything is wrapped in a try catch so I'm sure there was no issue with the PutObjectAsync operation.

I'm using Minio 3.1.12

**Comment by mia01** [comment]
I converted this to :
```
var documents = await _minioClient.ListObjectsAsync(AppIdentifier, "folder/").ToList();

```
That gives me consistent results every time. I still would like to know why the above didn't work though 🤔

**Comment by JonOdgaard** [comment]
I have been struggling with the same issue for at couple of hours
Your work around really saved my day

**Comment by molekm** [comment]
Same issue, while running Unit tests till fail I maybe will manage 3-4 times.
This issue is ongoing for almost a year for me (Minio on Docker).

**Comment by BigUstad** [comment]
Fixed with #524
